CLDR-18108 Give all languages a primary script: trivial cases (#4210)
unicode-org/cldr: docs/site/development/updating-codes/update-language-script-info/language-script-description.md

# Language Script Description

The [`language\_script.tsv`](https://github.com/unicode-org/cldr/blob/main/tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/language_script.tsv) data file should list all of the language / script combinations that are in common use. Usage by country is indicated in the [`country\_language\_population.tsv`](https://github.com/unicode-org/cldr/blob/main/tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/country_language_population.tsv) spreadsheet.

1. Every language should have 1 script considered the **primary** script.

>     1. This data is used to determine [the most Likely language and region](likelysubtags-and-default-content) so there needs to be at least 1 primary value.

Reviewer: ```suggestion
    1. This data is used to determine [the most Likely language and region](/development/updating-codes/likelysubtags-and-default-content) so there needs to be at least 1 primary value.
```

    2. __Changed in v47__ Include a primary script for historical languages (eg. Ancient Greek, Coptic). The primary script should reflect where the majority of the written corpus originates from.
2. Other scripts used for a language should be marked **secondary**.

Languages with multiple ambiguous scripts should have that reflected in their CLDR structure (eg. `sr_Cyrl_RS`), with aliases for the language\-region combinations.

> In order to re-generate the XML data use ConvertLanguageData as written about in [the article about updating the language scripts](.../update-language-script-info.md).

Reviewer: ```suggestion
In order to re-generate the XML data use ConvertLanguageData as written about in [the article about updating the language scripts](/development/updating-codes/update-language-script-info.md).
```

use site relative links (for now)

> In order to re-generate the XML data use ConvertLanguageData as written about in [the article about updating the language scripts](/development/update-language-script-info.md).

Reviewer: ```suggestion
In order to re-generate the XML data use ConvertLanguageData as written about in [the article about updating the language scripts](/development/updating-codes/update-language-script-info).
```

# Language Script Description

The [`language\_script.tsv`](https://github.com/unicode-org/cldr/blob/main/tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/language_script.tsv) data file should list all of the language / script combinations that are in common use. Usage by country is indicated in the [`country\_language\_population.tsv`](https://github.com/unicode-org/cldr/blob/main/tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/country_language_population.tsv) spreadsheet.

1. Every language needs at least 1 script considered the **primary** script.
    1. This data is used to determine [the most Likely language and region](likelysubtags-and-default-content) so there needs to be at least 1 primary value.
    2. [Changed in v47] Include a primary script for historical languages (eg. Ancient Greek, Coptic). The primary script should reflect where the majority of the written corpus originates from.

> 2. Languages written by significant populations with different scritps in different countries can have multiple **primary** scripts. The [likely subtags](https://www.unicode.org/cldr/charts/latest/supplemental/likely_subtags.html) patterns will use population counts to disambiguate the default script for each locale.

Reviewer 2: I disagree with this; I think it complicates our work. It is simplest if there is exactly 1 primary script, period, that will match the maximized value for the language. All others are not needed, because they will be derived from language population data. Eg if Kurdish has Latin as the primary script, but ku_Arab is the largest population (of ku*) speakers, then we will generate ku_IQ => ku_Arab_IQ in likely subtags.

Is there any other purpose to multiple primaries or having secondaries?

Author: I'm fine with that to (only ever 1 primary). Do you think the CLDR TC agrees with that? I don't think I had much involvement.

I was trying to strike a compromise with the concerns people had about picking a primary language (that was mentioned in the DDL group). I'll bring it up again in the next DDL meeting.

Reviewer: I agree that we should simply pick a primary, _just as_ we need to pick a single likely subtag (i.e. `zh-Hans`).  I know there are concerns, but structurally we just need to pick something.

Always having exactly 1 primary makes the most sense structurally.

The [`language\_script.tsv`](https://github.com/unicode-org/cldr/blob/main/tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/language_script.tsv) data file should list all of the language / script combinations that are in common use. Usage by country is indicated in the [`country\_language\_population.tsv`](https://github.com/unicode-org/cldr/blob/main/tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/country_language_population.tsv) spreadsheet.

1. Every language should have 1 script considered the **primary** script.

>     1. This data is used to determine [the most Likely language and region](/development/updating-codes/likelysubtags-and-default-content) so there needs to be at least 1 primary value.

Reviewer 2: ```suggestion
    1. This data is currently used to determine [the likely script for a language](/development/updating-codes/likelysubtags-and-default-content) so there needs to be at least 1 primary value. Because it is the default, it determines the script of locales without language codes in the `<territoryInfo>`. 
```

See also https://unicode-org.atlassian.net/browse/CLDR-18155?focusedCommentId=178748
